lib/xenops_server.ml

```diff
@@ -919,13 +919,23 @@ let is_no_sharept vgpu =
   | Some(_), Vgpu.Nvidia(_) -> true
   | _ -> false
 
+(* [is_nvidia_sriov vgpus pci] is true, if [pci] belongs to an SRIOV
+ * Nvidia vGPU in [vgpus] and false otherwise *)
+let is_nvidia_sriov vgpus pci =
+  let is_sriov = function
+  | Vgpu.{ implementation = Nvidia(_); virtual_pci_address = Some addr; _}
+      -> addr = pci.Pci.address
+  | _ -> false in
+  List.exists is_sriov vgpus
```

Comment:
Can a VM have both an Nvidia and AMD GPU?

Reply:
This predicate identifices PCI devices that need to be plugged early and not announced to QEMU. It would be `false` for an AMD GPU - as it should, I believe.